test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConditionalCreateTests.cs

```diff
@@ -0,0 +1,99 @@
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAResource_WhenCreatingConditionallyWithNoIdAndNoExisting_TheServerShouldReturnTheResourceSuccessfully()
+        {
+            var observation = Samples.GetDefaultObservation().ToPoco<Observation>();
+            observation.Id = null;
+
+            FhirResponse<Observation> updateResponse = await Client.CreateAsync(
+                observation,
+                $"identifier={Guid.NewGuid().ToString()}");
+
+            Assert.Equal(HttpStatusCode.Created, updateResponse.StatusCode);
+
+            Observation updatedResource = updateResponse.Resource;
+
+            Assert.NotNull(updatedResource);
+            Assert.NotNull(updatedResource.Id);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAResourceWithNoId_WhenCreatingConditionallyWithOneMatch_TheServerShouldReturnOK()
+        {
+            var observation = Samples.GetDefaultObservation().ToPoco<Observation>();
+            var identifier = Guid.NewGuid().ToString();
+
+            observation.Identifier.Add(new Identifier("http://e2etests", identifier));
+            FhirResponse<Observation> response = await Client.CreateAsync(observation);
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+            var observation2 = Samples.GetDefaultObservation().ToPoco<Observation>();
+
+            FhirResponse<Observation> updateResponse = await Client.CreateAsync(
+                observation2,
+                $"identifier={identifier}");
+
+            Assert.Equal(HttpStatusCode.OK, updateResponse.StatusCode);
+
+            Observation updatedResource = updateResponse.Resource;
+
+            Assert.Null(updatedResource);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAResource_WhenCreatingConditionallyWithMultipleMatches_TheServerShouldFail()
+        {
+            var observation = Samples.GetDefaultObservation().ToPoco<Observation>();
+            var identifier = Guid.NewGuid().ToString();
+
+            observation.Identifier.Add(new Identifier("http://e2etests", identifier));
+
+            FhirResponse<Observation> response = await Client.CreateAsync(observation);
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+            FhirResponse<Observation> response2 = await Client.CreateAsync(observation);
+            Assert.Equal(HttpStatusCode.Created, response2.StatusCode);
+
+            var observation2 = Samples.GetDefaultObservation().ToPoco<Observation>();
+            observation2.Id = Guid.NewGuid().ToString();
+
+            var exception = await Assert.ThrowsAsync<FhirException>(() => Client.CreateAsync(
+                observation2,
+                $"identifier={identifier}"));
+
+            Assert.Equal(HttpStatusCode.PreconditionFailed, exception.Response.StatusCode);
+        }
+    }
```

Comment:
Would it be useful to add a test where we add the If-None-Exist header with an empty value?